lib/icingadb/icingadb-utility.cpp

```diff
@@ -245,6 +245,17 @@ String IcingaDB::GetLowerCaseTypeNameDB(const ConfigObject::Ptr& obj)
 }
 
 long long IcingaDB::TimestampToMilliseconds(double timestamp) {
+	// In addition to the limits of the Icinga DB MySQL (0 - 2^64) and PostgreSQL (0 - 2^63) schemata,
+	// years not fitting in YYYY may cause problems, see e.g. https://github.com/golang/go/issues/4556.
+	// RFC 3339: "All dates and times are assumed to be (...) somewhere between 0000AD and 9999AD."
+	//
+	// The below limits include "better safe than sorry" buffers
+	// not to worry about time zones, floating point arithmetic details and the precise numbers themselves.
+	// The buffers were chosen so that the numbers below are easy to write and read.
+	if (timestamp <= 0.0 || timestamp > 253'402e6 /* 9999-12-28 */) {
```

Comment:
How is `253'402e6` easy to read[^1]? I'd say the most common way to read a Unix timestamp is to paste it into some tool that converts it into a human-readable form. Most of these tools will fail to directly read the C++ notation you're using here, requiring an extra conversion step.

When actually doing this, you'll notice out that the comment is somewhat misleading, as in fact, the constant represents a more or less random time on that day:

```console
$ date -ud @253402000000
Tue 28 Dec 9999 12:26:40 UTC
```

How can it be so hard to just write a plain number?

[^1]: I can't say about writing from your perspective, but I can't imagine it has been less work than running something like `date -ud '9999-12-31 00:00:00' +%s` and just using the resulting value as-is.